### PR TITLE
Refactor some stringly typed route-creation code

### PR DESCRIPTION
**Pull request by tipsy**

### CODE REVIEW

**Comment by MouettE-SC** [comment]
That's actually an idea I had when renaming the afterafter stuff (and run into a case problem). Didn't go with it in order to get the merge quickly but it's better this way. 👍

File: src/main/java/spark/RouteImpl.java [hunk 1]
```diff
@@ -33,6 +33,17 @@
     private Object delegate;
 
     /**
+     * Prefix the path (used for {@link Service#path}
```
Comment (pkkummermo) [inline src/main/java/spark/RouteImpl.java]: Close parenthesis <3

File: src/main/java/spark/route/Routes.java [hunk 2]
```diff
@@ -51,34 +53,21 @@ protected Routes() {
     }
 
     /**
-     * Parse and validates a route and adds it
-     *
-     * @param route      the route path
-     * @param acceptType the accept type
-     * @param target     the invocation target
+     * Add a route
```
Comment (pkkummermo) [inline src/main/java/spark/route/Routes.java]: Add newline

File: src/main/java/spark/route/Routes.java [hunk 3]
```diff
@@ -51,34 +53,21 @@ protected Routes() {
...
-                                  + ".");
-                return;
-            }
-            addRoute(method, url, acceptType, target);
-        } catch (Exception e) {
-            LOG.error("The @Route value: " + route + " is not in the correct format", e);
-        }
+    /**
+     * Add a filter
+     * @param httpMethod the http-method of the route
```
Comment (pkkummermo) [inline src/main/java/spark/route/Routes.java]: Add newline

File: src/test/java/spark/route/RoutesTest.java [hunk 4]
```diff
@@ -1,54 +0,0 @@
...
-
-        Routes simpleRouteMatcher = Routes.create();
-        simpleRouteMatcher.add(route, acceptType, target);
-
-        //then
-        List<RouteEntry> routes = Whitebox.getInternalState(simpleRouteMatcher, "routes");
-        assertEquals("Should return 0 because test is not a valid http method, so the route is not added to the list",
-                     routes.size(), 0);
-    }
-}
```
Comment (pkkummermo) [inline src/test/java/spark/route/RoutesTest.java]: Any new tests which should be added in their place?

File: src/main/java/spark/FilterImpl.java [hunk 5]
```diff
@@ -34,6 +34,17 @@
     private Filter delegate;
 
     /**
+     * Prefix the path (used for {@link Service#path}
```
Comment (tipsy) [inline src/main/java/spark/FilterImpl.java]: Done!

File: src/main/java/spark/RouteImpl.java (hunk 1 above)
Comment (tipsy) [inline src/main/java/spark/RouteImpl.java]: Done!

File: src/main/java/spark/route/Routes.java (hunk 2 above)
Comment (tipsy) [inline src/main/java/spark/route/Routes.java]: Done!

File: src/main/java/spark/route/Routes.java (hunk 3 above)
Comment (tipsy) [inline src/main/java/spark/route/Routes.java]: Done!

File: src/test/java/spark/route/RoutesTest.java (hunk 4 above)
Comment (tipsy) [inline src/test/java/spark/route/RoutesTest.java]: Not really. This parsing is removed, adding routes is covered by other tests.

**Pull request merged**